client/lib/utils/coerce-values.js

```diff
@@ -0,0 +1,85 @@
+import coerce from 'type-coerce';
+
+/**
+ * Retrieve a field's schema, handling referenced schema definitions if need be.
+ *
+ * @param {Object} fieldSchema - Schema object for field.
+ * @param {Object} definitions - Common definitions.
+ * @returns {Object} - Schema object for field, potentially pulled from $ref's definition.
+ */
+const getFieldSchema = ( fieldSchema, definitions ) => {
+	if ( fieldSchema.hasOwnProperty( '$ref' ) ) {
+		const definitionKey = fieldSchema.$ref.match( /^#\/definitions\/(.+)/ );
```

Comment:
I think `fieldSchema.$ref.split('#/definitions/')[1]` would be cleaner (but yet again, I hate RegExps too much, so feel free to ignore me)